src/routes/transactions/helpers/kiln-native-staking.helper.spec.ts

```diff
@@ -0,0 +1,245 @@
+import { KilnDecoder } from '@/domain/staking/contracts/decoders/kiln-decoder.helper';
+import { StakingRepository } from '@/domain/staking/staking.repository';
+import { KilnNativeStakingHelper } from '@/routes/transactions/helpers/kiln-native-staking.helper';
+import { TransactionFinder } from '@/routes/transactions/helpers/transaction-finder.helper';
+import { faker } from '@faker-js/faker';
+import { getAddress } from 'viem';
+
+const mockStakingRepository = jest.mocked({
+  getStakes: jest.fn(),
+} as jest.MockedObjectDeep<StakingRepository>);
+
+describe('KilnNativeStakingHelper', () => {
+  let target: KilnNativeStakingHelper;
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+
+    const multiSendDecoder = new MultiSendDecoder();
+    const transactionFinder = new TransactionFinder(multiSendDecoder);
+    target = new KilnNativeStakingHelper(
+      transactionFinder,
+      mockStakingRepository,
+    );
+  });
+
+  describe.skip('findDepositTransaction', () => {
+    it('should return a `deposit` transaction', () => {});
+
+    it('should return a batched `deposit` transaction', () => {});
+
+    it('should return null if a `deposit` transaction is not from a known staking contract', () => {});
+
+    it('should return null if a batched `deposit` transaction is not from a known staking contract', () => {});
+
+    it('should return null if the transaction is not a `deposit` transaction', () => {});
+
+    it('should return null if the transaction batch contains no `deposit` transaction', () => {});
+  });
+
+  describe.skip('findValidatorsExitTransaction', () => {
+    it('should return a `requestValidatorsExit` transaction', () => {});
+
+    it('should return a batched `requestValidatorsExit` transaction', () => {});
+
+    it('should return null if a `requestValidatorsExit` transaction is not from a known staking contract', () => {});
+
+    it('should return null if a requestValidatorsExit `deposit` transaction is not from a known staking contract', () => {});
+
+    it('should return null if the transaction is not a `requestValidatorsExit` transaction', () => {});
+
+    it('should return null if the transaction batch contains no `requestValidatorsExit` transaction', () => {});
+  });
+
+  describe.skip('findWithdrawTransaction', () => {
+    it('should return a `batchWithdrawCLFee` transaction', () => {});
+
+    it('should return a batched `batchWithdrawCLFee` transaction', () => {});
+
+    it('should return null if a `batchWithdrawCLFee` transaction is not from a known staking contract', () => {});
+
+    it('should return null if a batchWithdrawCLFee `deposit` transaction is not from a known staking contract', () => {});
+
+    it('should return null if the transaction is not a `batchWithdrawCLFee` transaction', () => {});
+
+    it('should return null if the transaction batch contains no `batchWithdrawCLFee` transaction', () => {});
+  });
```

Comment:
This will be added in a follow up as they are out of scope.

Reply:
Very nit: what do you think about setting them as "todos" instead of skipping the blocks?

Something like:
```typescript
describe('findDepositTransaction', () => {
    it.todo('should return a `deposit` transaction');
```

Reply:
Changed in 055fed1dc06cd51564316680a6c5fb42af0a8f1b.